zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/CreateProcessInstanceWithTerminationInstructionTest.java

```diff
@@ -195,4 +195,83 @@ public void shouldNotTerminateWhenNonInterruptingBoundaryEventIsActivated() {
         .doesNotContain(Tuple.tuple(ProcessInstanceIntent.ELEMENT_TERMINATING, processId))
         .doesNotContain(Tuple.tuple(ProcessInstanceIntent.ELEMENT_TERMINATED, processId));
   }
+
+  @Test
+  public void shouldTerminateWhenElementIsInNestedEmbeddedSubprocess() {
+    // given
+    final String processId = "process";
+    final String elementBeforeTermination = "element";
+    final String boundaryEventNestedSubProcessEnd = "boundaryNestedSubProcessEnd";
+    final String boundaryEventId = "boundary";
+    ENGINE
+        .deployment()
+        .withXmlResource(
+            Bpmn.createExecutableProcess(processId)
+                .startEvent()
+                .subProcess("firstSubProcess")
+                .embeddedSubProcess()
+                .startEvent()
+                .subProcess("nestedSubProcess")
+                .embeddedSubProcess()
+                .startEvent()
+                .serviceTask(elementBeforeTermination, t -> t.zeebeJobType("jobType"))
+                .subProcessDone()
+                .boundaryEvent("boundary")
+                .cancelActivity(true)
+                .message(
+                    messageBuilder ->
+                        messageBuilder
+                            .name("myMessage")
+                            .zeebeCorrelationKey("=\"myCorrelationKey\""))
+                .endEvent(boundaryEventNestedSubProcessEnd)
+                .moveToActivity("nestedSubProcess")
+                .endEvent("nestedSubProcessEnd")
+                .subProcessDone()
+                .endEvent()
+                .done())
+        .deploy();
+
+    final long processInstanceKey =
+        ENGINE
+            .processInstance()
+            .ofBpmnProcessId(processId)
+            .withRuntimeTerminateInstruction(elementBeforeTermination)
+            .create();
+
+    // when
+    ENGINE
+        .message()
+        .withName("myMessage")
+        .withCorrelationKey("myCorrelationKey")
+        .withTimeToLive(1000)
+        .publish();
```

Comment:
🔧 In tests, messages are published with a default TTL already, so no need to specify one explicitly. If we ever need a different TTL in tests, we can adjust the default rather than explicit values in all tests.